gpytorch/module.py

```diff
@@ -74,7 +74,7 @@ def initialize(self, **kwargs):
                 module.initialize(**{name: val})
             elif not hasattr(self, name):
                 raise AttributeError("Unknown parameter {p} for {c}".format(p=name, c=self.__class__.__name__))
-            elif name not in self._parameters:
+            elif name not in self._parameters and name not in self._buffers:
```

Comment:
@gpleiss so this actually breaks some of the things we're doing to re-initalize models with new data. Say you have a model using a `FixedNoiseGaussianLikelihood` with `n` initial observations. You get another observation in a BayesOpt loop, and you want to update your model in-place (using set_train_data). For this you also need to update the new noise value. But if you go through `initialize` (which is what this change here results in) you'll end up with a runtime error b/c the `initialize` method tries to reshape a 11 element tensor to a 10 element tensor. 

The alternative is to re-register the buffer itself with the larger tensor. But that will mean that this has to be done in a model-specific way and we need to know the underlying model setup (rather than using a standard noise property).

Thoughts?

Reply:
I'm not a big fan of `initialize` being able to change the shape of parameters because I feel like that's going to result in more bugs than legitimate use cases. I'd rather FixedNoiseGaussianLikelihood have functionality for this -- for example the `fantasy` method that will be used for adding noises to the vector.

Reply:
Yeah not suggesting that initialize itself should be able to change the size. I guess for now we can handle this on our end until we have the fantasy method functionality set up.

Reply:
It seems that this will be an issue mostly related to fantasy/BayesOpt updates with FixedNoiseGaussianLikelihood. We should maybe brainstorm interfaces specifically for this case.